tests/CMakeLists.txt

```diff
@@ -60,6 +60,19 @@ add_test(NAME repro_avif_decode_fuzzer COMMAND repro_avif_decode_fuzzer
 ################################################################################
 # GoogleTest
 
+# Adds a gtest from file TEST_NAME.cc located in the gtest folder. Extra arguments
+# are considered as extra linked libraries.
+macro(add_avif_gtest TEST_NAME USE_DATA)
+    add_executable(${TEST_NAME} gtest/${TEST_NAME}.cc)
+    target_include_directories(${TEST_NAME} PRIVATE ${GTEST_INCLUDE_DIRS})
+    target_link_libraries(${TEST_NAME} PRIVATE aviftest_helpers ${GTEST_BOTH_LIBRARIES} ${ARGN})
```

Comment:
I am surprised this works everywhere: `gtest_main` should not be linked in tests already defining a custom `main()`.

Reply:
Explained here: https://github.com/google/googletest/issues/2674

Reply:
It would be good to avoid linking tests defining its own `main()` function with `gtest_main`.

Reply:
Exactly, that is why the macto add_avif_gtest_with_data was created below which links against `GTEST_LIBRARIES` and not `GTEST_BOTH_LIBRARIES`.
For now, tests defining `main` are called with a data argument.
If needs be for tests with `main` but different arguments, we can create a third macro.

Reply:
Thank you for the explanation. I didn't know the only reason our tests need to define their own main() function is to take a data argument. The current two macros are good.